python/defs.bzl

```diff
@@ -41,6 +41,45 @@ def _add_tags(attrs):
         attrs["tags"] = [_MIGRATION_TAG]
     return attrs
 
+def _current_py_toolchain_impl(ctx):
+    toolchain = ctx.toolchains[ctx.attr._toolchain]
+
+    files = []
+    vars = {}
+
+    if toolchain.py3_runtime and toolchain.py3_runtime.interpreter:
+        files += [toolchain.py3_runtime.interpreter] + toolchain.py3_runtime.files.to_list()
+        vars = dict(vars.items() + {"PYTHON3": toolchain.py3_runtime.interpreter.path}.items())
+
+    if toolchain.py2_runtime and toolchain.py2_runtime.interpreter:
+        files += [toolchain.py2_runtime.interpreter] + toolchain.py2_runtime.files.to_list()
+        vars = dict(vars.items() + {"PYTHON2": toolchain.py2_runtime.interpreter.path}.items())
+
+    return [
+        toolchain,
+        platform_common.TemplateVariableInfo(vars),
+        DefaultInfo(
+            runfiles = ctx.runfiles(files = files),
+            files = depset(files),
+        ),
+    ]
+
+current_py_toolchain = rule(
+    doc = """
+    This rule exists so that the current python toolchain can be used in the `toolchains` attribute of
+    other rules, such as genrule. It allows exposing a python toolchain after toolchain resolution has
+    happened, to a rule which expects a concrete implementation of a toolchain, rather than a
+    toolchain_type which could be resolved to that toolchain.
+    """,
+    implementation = _current_py_toolchain_impl,
+    attrs = {
+        "_toolchain": attr.string(default = str(Label("@bazel_tools//tools/python:toolchain_type"))),
+    },
+    toolchains = [
+        str(Label("@bazel_tools//tools/python:toolchain_type")),
```

Comment:
Every time I see this, I wish we had a const for the toolchain type in the same way we have https://github.com/bazelbuild/bazel/blob/49610123de3c555f1f29dd4c6d91101c659a26b1/tools/cpp/toolchain_utils.bzl#L20.

Reply:
I can add such a constant changes change all references to `"@bazel_tools//tools/python:toolchain_type"` if you like, although that should probably be in a separate PR

Reply:
Up to you if you want to contribute further.